test/fixtures/v2VariableManyConditionsComplex.js

```diff
@@ -0,0 +1,210 @@
+                    },
+                    Match: {
+                        Comparator: 'ge',
+                        Type: 'GreaterEqualsValue'
+                    },
+                    Type: 'SpamTest'
+                }
+            ],
+            Type: 'AllOf'
+        },
+        Then: [
+            {
+                Type: 'Return'
+            }
+        ],
+        Type: 'If'
+    },
+    {
+        Text:
+            '/**\n * @type or\n * @comparator contains\n * @comparator contains\n * @comparator default\n * @comparator is\n */',
+        Type: 'Comment'
+    },
+    {
+        If: {
+            Tests: [
+                {
+                    Headers: ['Subject'],
+                    Keys: ['jeanne', '/monique\\d+/'],
+                    Match: {
+                        Type: 'Contains'
+                    },
+                    Format: {
+                        Type: 'UnicodeCaseMap'
+                    },
+                    Type: 'Header'
+                },
+                {
+                    Headers: ['From'],
+                    Keys: ['protonmail'],
+                    Match: {
+                        Type: 'Contains'
+                    },
+                    Format: {
+                        Type: 'UnicodeCaseMap'
+                    },
+                    AddressPart: {
+                        Type: 'All'
+                    },
+                    Type: 'Address'
+                },
+                {
+                    Headers: ['X-Attached'],
+                    Type: 'Exists'
+                },
+                {
+                    Headers: ['To', 'Cc', 'Bcc'],
+                    Keys: [
+                        {
+                            Value: '${frommail}',
+                            Type: 'VariableString'
+                        }
+                    ],
+                    Match: {
+                        Type: 'Is'
+                    },
+                    Format: {
+                        Type: 'UnicodeCaseMap'
+                    },
+                    AddressPart: {
+                        Type: 'All'
+                    },
+                    Type: 'Address'
+                }
+            ],
+            Type: 'AnyOf'
+        },
+        Then: [
+            {
+                Name: 'archive',
+                Type: 'FileInto'
+            },
+            {
+                Name: 'polo',
+                Type: 'FileInto'
+            },
+            {
+                Flags: ['\\Seen', '\\Flagged'],
+                Type: 'AddFlag'
+            },
+            {
+                Type: 'Keep'
+            },
+            {
+                Message: "<div>Je mange une pomme, déso pas déso mais j'ai faim pedro</div> ",
+                Args: {
+                    MIMEType: 'text/html'
+                },
+                Type: 'Vacation'
+            }
+        ],
+        Type: 'If'
+    }
+];
+
+const simple = {
```

Comment:
No, simple should be `{}`. 
The tree above is not a graphical filter, it's only a regular filter. The node [here](https://github.com/ProtonMail/sieve.js/pull/11/files#diff-7c473c465d1035cd53ccc2a778bf5776R14) does not belong  to the generated filter.